generators/php/model/src/union/UnionGenerator.ts

```diff
@@ -0,0 +1,1329 @@
+        variant: SingleUnionType,
+        variableGetter: php.CodeBlock,
+        type: php.Type
+    ): php.CodeBlock | null {
+        const typeCheck = this.getTypeCheck(variableGetter, type);
+
+        if (typeCheck == null) {
+            return null;
+        }
+
+        const negation = php.codeblock((writer) => {
+            writer.write("!");
+            writer.write("(");
+            writer.writeNode(this.isMethodCheck(variant));
+            writer.write(")");
+        });
+
+        return php.codeblock((writer) => {
+            writer.controlFlow("if", negation);
+            writer.writeNodeStatement(
+                this.getErrorThrow(this.getVariantTypeCheckErrorMessage(variant, variableGetter))
+            );
+            writer.endControlFlow();
+        });
+    }
+
+    private isMethodCheck(variant: SingleUnionType): php.CodeBlock {
+        const discriminantCheck = this.getDiscriminantCheck(this.discriminantGetter(), variant);
+        const typeCheck = this.getTypeCheck(this.valueGetter(), this.getReturnType(variant));
+
+        return php.codeblock((writer) => {
+            if (typeCheck) {
+                writer.writeNode(typeCheck);
+                writer.write("&& ");
+            }
+
+            writer.writeNode(discriminantCheck);
+        });
+    }
+
+    private getErrorThrow(errorMessage: php.CodeBlock): php.CodeBlock {
+        return php.codeblock((writer) => {
+            writer.writeNode(
+                php.throwException({
+                    classReference: this.context.getExceptionClassReference(),
+                    arguments_: [
+                        php.codeblock((writer) => {
+                            writer.writeNode(errorMessage);
+                        })
+                    ],
+                    multiline: true
+                })
+            );
+        });
+    }
+
+    private getVariantTypeCheckErrorMessage(variant: SingleUnionType, variableGetter: php.CodeBlock): php.CodeBlock {
+        return php.codeblock((writer) => {
+            writer.write('"');
+            writer.write("Expected ");
+            writer.write(variant.discriminantValue.wireValue);
+            writer.write("; got ");
+            writer.write('"');
+            writer.write(" . ");
+            writer.writeNode(this.discriminantGetter());
+            writer.write(" . ");
+            writer.write('"');
+            writer.write("with value of type ");
+            writer.write('"');
+            writer.write(" . ");
+            writer.writeNode(
+                php.invokeMethod({
+                    method: "get_debug_type",
+                    arguments_: [variableGetter],
+                    static_: true
+                })
+            );
+        });
+    }
+
+    private getDeserializationArrayKeyExistsErrorMessage(propertyName: NameAndWireValue): php.CodeBlock {
+        return php.codeblock(`"Json data is missing property '${propertyName.wireValue}'"`);
```

Comment:
nit: Let's use `JSON` here
```suggestion
        return php.codeblock(`"JSON data is missing property '${propertyName.wireValue}'"`);
```